src/lava/proc/scif/models.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import numpy as np
+
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyInPort, PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.proc.scif.process import SCIF
+
+
+@implements(proc=SCIF, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('fixed_pt')
+class PySCIFModelFixed(PyLoihiProcessModel):
+    """Fixed point implementation of Stochastic Constraint Integrate and
+    Fire (SCIF) neuron.
+    """
+    a_in = LavaPyType(PyInPort.VEC_DENSE, int, precision=8)
+    s_sig_out = LavaPyType(PyOutPort.VEC_DENSE, int, precision=8)
+    s_wta_out = LavaPyType(PyOutPort.VEC_DENSE, int, precision=8)
+
+    u: np.ndarray = LavaPyType(np.ndarray, int, precision=24)
+    v: np.ndarray = LavaPyType(np.ndarray, int, precision=24)
+    beta: np.ndarray = LavaPyType(np.ndarray, int, precision=8)
+
+    bias: np.ndarray = LavaPyType(np.ndarray, int, precision=24)
+    theta: np.ndarray = LavaPyType(np.ndarray, int, precision=24)
+    neg_tau_ref: np.ndarray = LavaPyType(np.ndarray, int, precision=24)
+    enable_noise: np.ndarray = LavaPyType(np.ndarray, int, precision=1)
+
+    def run_spk(self) -> None:
+        # Receive synaptic input
+        a_in_data = self.a_in.recv()
+
+        # Define spike vectors
+        s_sig = np.zeros_like(self.v)
+        s_wta = np.zeros_like(self.v)
+
+        # Saturated add the incoming activation
+        self.u += a_in_data
+        self.u[self.u > 2 ** 23 - 1] = 2 ** 23 - 1
+        self.u[self.u < -2 ** 23] = -2 ** 23
+
+        # Populate the buffer for local computation
+        lsb = self.beta.copy()
+        lsb &= 3
+        self.beta <<= 2
+        self.beta[self.beta >= 256] = 0  # Overflow 8-bit unsigned beta to 0
+
+        # Gather spike and unsatisfied indices for summation axons
```

Comment:
Could you give a bit more detail in this comment? I'm not fully sure why this is being done. 
Alternatively, you may want to add the full equation to the Process docstring.
Thanks